### ISSUE TITLE
JavaScript mode scope highlighting doesn't adjust for let/const block scopes

### ISSUE THREAD

**Issue by tmcw** [codemirror/CodeMirror#5138]
Here's an example:

![2017-12-20 at 1 15 pm](https://user-images.githubusercontent.com/32314/34229112-f3db9d46-e587-11e7-8111-5aa394e82687.png)

```js
function findSequence(goal) {
  {
    let aBlockScopedVar = 10;
  }
  return aBlockScopedVar;
}
```

Kind of a nitpicky issue, but `aBlockScopedVar` is highlighted as `variable-2` in this case, when it should be highlighted as `variable`, because the block containing the `let` binding contains the binding.

As far as I can tell, fixing this would require context objects to have a 'type' - there'd be 'function contexts' and 'block contexts': `var` bindings would go into `register` and get registered to the nearest function context if there is one, and `let` or `const` bindings would attach themselves to the nearest context of either type.

**Comment by marijnh** [comment]
This doesn't greatly bother me, but a pull request that fixes it would be welcome.

**Comment by tmcw** [comment]
Fixed in related commit to #5296!
